### PR TITLE
Created Confusion matrix function as well as unit test functions

### PR DESCRIPTION
**Pull request by miketianchen**
should solve #20

### CODE REVIEW

File: R/confusion_matrix.R [hunk 1]
```diff
@@ -2,20 +2,66 @@
 #' visual. If predicted_y array is passed in,other evaluation scoring metrics such as
 #' Recall, and precision will also be produced
```
Comment (anasm-17) [inline R/confusion_matrix.R:3]: Does it only work for binary classification? if so it is worth mentioning the limitations

File: R/confusion_matrix.R [hunk 2]
```diff
@@ -2,20 +2,66 @@
...
+    ggplot2::scale_fill_manual(values = c(good = "green", bad = "red")) +
+    ggplot2::theme_bw() +
+    ggplot2::xlim(rev(levels(table$Reference))) +
+    ggplot2::theme(legend.position = "none") +
+    ggplot2::labs(title = title)
+
+  metric_score <- broom::tidy(confusion_matrix$byClass)
+
+  print(confusion_plot)
+  return(metric_score)
```
Comment (anasm-17) [inline R/confusion_matrix.R:64]: I think it maybe worth returning the table as well.
```suggestion
  return(list(cf_table = table, metric = metric_score))
```

**Review by anasm-17 — APPROVED** [review]
See changes for future improvement :) looks good otherwise

File: R/confusion_matrix.R (hunk 2 above)
Comment (miketianchen) [reply R/confusion_matrix.R:64]: image


that is the table.

File: R/confusion_matrix.R (hunk 1 above)
Comment (miketianchen) [reply R/confusion_matrix.R:3]: image

works for more than 2 classes

**Pull request merged**